src/pages/Defi/hooks/useFoxyBalances.tsx

```diff
@@ -45,57 +46,65 @@ export type UseFoxyBalancesReturn = {
   loading: boolean
 }
 
+const moduleLogger = logger.child({
+  namespace: ['DeFi', 'Hooks', 'UseFoxyBalances'],
+})
+
 async function getFoxyOpportunities(
   balances: PortfolioBalancesById,
   api: FoxyApi,
   userAddress: string,
 ) {
   const acc: Record<string, FoxyOpportunity> = {}
-  const opps = await api.getFoxyOpportunities()
-  for (let index = 0; index < opps.length; index++) {
-    // TODO: assetIds in vaults
-    const opportunity = opps[index]
-    const withdrawInfo = await api.getWithdrawInfo({
-      contractAddress: opportunity.contractAddress,
-      userAddress,
-    })
-    const rewardTokenAssetId = toAssetId({
-      chain: opportunity.chain,
-      network: NetworkTypes.MAINNET,
-      assetNamespace: 'erc20',
-      assetReference: opportunity.rewardToken,
-    })
-    const contractAssetId = toAssetId({
-      chain: opportunity.chain,
-      network: NetworkTypes.MAINNET,
-      assetNamespace: 'erc20',
-      assetReference: opportunity.contractAddress,
-    })
-    const tokenAssetId = toAssetId({
-      chain: opportunity.chain,
-      network: NetworkTypes.MAINNET,
-      assetNamespace: 'erc20',
-      assetReference: opportunity.stakingToken,
-    })
-    const balance = balances[rewardTokenAssetId]
-
-    const pricePerShare = api.pricePerShare()
-    acc[opportunity.contractAddress] = {
-      ...opportunity,
-      balance: bnOrZero(balance).toString(),
-      contractAssetId,
-      tokenAssetId,
-      rewardTokenAssetId,
-      pricePerShare: bnOrZero(pricePerShare),
-      withdrawInfo,
+  try {
+    const opportunities = await api.getFoxyOpportunities()
+    for (let index = 0; index < opportunities.length; index++) {
```

Comment:
a `for/or` would work just fine here. we should avoid these types of indexed for loops due to potential for logic errors.